.github/workflows/build.yml

```diff
@@ -317,3 +320,44 @@ jobs:
       run: make pythoninfo
     - name: Tests
       run: xvfb-run make buildbottest TESTOPTS="-j4 -uall,-cpu"
+
+  check:  # This job does nothing and is only used for the branch protection
+    if: always()
+
+    needs:
+    - check_source  # Transitive dependency, needed to access `run_tests` value
+    - check_generated_files
+    - build_win32
+    - build_win_amd64
+    - build_macos
+    - build_ubuntu
+    - build_ubuntu_ssltests
+    - build_asan
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Decide whether the needed jobs succeeded or failed
+      uses: re-actors/alls-green@198badcb65a1a44528f27d5da555c4be9f12eac6
+      with:
+        allowed-skips: >-
```

Comment:
This setting tells the action that if the listed jobs (comma-separated) got skipped, that's okay. But the jobs are allowed to be skipped based on the same condition they've got set. If the first job requests test runs, this list will compute as empty and none of the jobs will be allowed to be skipped (meaning that skips would turn into failures).

Reply:
Do I understand this right?

- If `check-docs` starts, it must pass; if they skip, that's fine.

- If `check_generated_files` or any of the other jobs in its group start, they must all pass; if they all skip, that's fine.

- If `test_hypothesis` starts, it must pass; if they skip, that's fine.

Reply:
@hugovk almost: this is paired with the setting above :point_up: — essentially, if something is allowed to be skipped, and it is skipped, the action will treat it as non-failure.
But if something is allowed to be skipped and is not skipped, whether its failure is to be treated as such is controlled by the `allowed-failures` input.

Also, not they are not grouped inside the action, the templating here just produces a comma-separated list of things and the action doesn't know it was in some group externally. So yes, such a relation kinda exists, but I would say that it's indirect.

And since there's no “groups”, if any of the jobs that are allowed to be skipped run, each individual job is evaluated separately from the “group”, based on the fact of failures being allowed or not.

N.B. Whether a job is allowed to fail is also contributed by the `continue-on-error` setting, which is mostly useful for controlling individual matrix jobs. If a job was allowed to fail prior to this one, the action will not “see” if it was red and treat it as green.

Reply:
> * If `test_hypothesis` starts, it must pass; if they skip, that's fine.

With the context I added earlier, seeing that `test_hypothesis` is listed in `allowed-failures` unconditionally, it will never cause failures. This reflects the current branch protection configuration as seen in the Checks widget in PRs.